src/HTML5/Parser/Tokenizer.php

```diff
@@ -796,7 +787,7 @@ protected function doctype()
             if (false === $id) {
                 $this->events->doctype($doctypeName, $type, $pub, false);
 
-                return false;
+                return true;
```

Comment:
I changed that one, because I thin the previous code was wrong. AFAIU, the methods were meant to return `true` when the consumed some of the input, which happens in that case.

I don't think that case is covered by tests though (at least not with some content after it).